src/storage/query/QueryBaseProcessor.h

```diff
@@ -34,6 +37,32 @@ struct Bucket {
 
 using OneVertexResp = std::tuple<PartitionID, VertexID, kvstore::ResultCode>;
 
+class ReservoirSampling final {
+public:
+    ReservoirSampling() {
+        rng_.seed(nebula::time::WallClock::fastNowInMicroSec());
+    }
+
+    int64_t sampling() {
```

Comment:
I suggest pass the samples total number N in ctor,  put the final results vector in this class. 
And as a common class,  it should be a template class.  The class looks like:

```
template<class T>
class ReservoirSampling final {
public:
    ReservoirSampling(int32_t N) {
        rng_.seed(nebula::time::WallClock::fastNowInMicroSec());
        num_ = N;   
    }

    int64_t sampling(T&&  element) {
           ....
    }

    
private:
       std::vector<T>   samples_;
       int32_t num_;
       ....
```



At last, add a UT for ReservoirSampling,  add we'd better move it into common dir.

Reply:
I'll do it.